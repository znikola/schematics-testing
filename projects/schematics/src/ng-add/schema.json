{
  "$schema": "http://json-schema.org/schema",
  "$id": "Schematics",
  "title": "Testing Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    }
  },
  "required": []
}
