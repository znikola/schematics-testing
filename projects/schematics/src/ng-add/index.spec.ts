import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import * as path from 'path';
import { Schema as SchematicsOptions } from './schema';

const collectionPath = path.join(__dirname, '../collection.json');

describe('ng add @schematics/test', () => {
  const schematicRunner = new SchematicTestRunner(
    '@schematics/test',
    collectionPath
  );

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const defaultOptions: SchematicsOptions = {
    project: 'schematics-test',
  };

  beforeEach(async () => {
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
  });

  it('should create the file', async () => {
    appTree = await schematicRunner
      .runSchematicAsync('ng-add', defaultOptions, appTree)
      .toPromise();

    expect(appTree.readContent('new-file.ts')).toMatchSnapshot();
  });
});
