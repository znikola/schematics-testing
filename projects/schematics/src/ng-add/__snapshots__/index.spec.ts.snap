// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ng add @test/schematics should create the file 1`] = `"console.log('ðŸ‘‹ from schematics')"`;
